src/heap.js

```diff
@@ -78,22 +78,6 @@ class Heap {
     );
   }
 
-  /**
-   * Compares parent & child nodes
-   * and returns true if they are in right positions
-   *
-   * @private
-   * @param {object|number|string} parent
-   * @param {object|number|string} child
-   * @returns {boolean}
-   */
-  _compareByIndex(parentIndex, childIndex) {
-    return this._compareKeys(
```

Comment:
It took a bit a while for me to understand how this code works since `__compareKeys` is not defined in the Heap class.
Do you think we can accept `_compareKeys` as a constructor argument, and throw an error if not provided.

And MaxHeap and MinHeap can have their own default `_compareKeys` function and pass it to Heap in it.